### ISSUE TITLE
Improve ForbiddenImport docs and possibly behavior

### ISSUE THREAD

**Issue by ColtonIdle** [detekt/detekt#6088]
## Expected Behavior
ForbiddenImports may just be the easiest way for detekt to be used in a project. A lot of us have projects where we'd like to forbid the use of some class. 

## Context
Some ideas I have

1. Docs say
Configuration options:
imports (default: [])

imports which should not be used

forbiddenPatterns (default: '')

reports imports which match the specified regular expression. For example net.*R.

But it'd be nice for a legit usage to be shown like

imports:
  - value: 'a.b.c.Yes'
     reason: 'Use this instead of that'

2. I think `forbiddenPatterns (default: '') ` is a bad name. patterns is plural. but it only takes a single pattern.

3. Some more docs on wildcard usage etc would be nice instead of trial and error when adding the rule.

4. The noncompliant code section is confusing. It doesn't actually relate to the above example? Unless im missing something.

**Comment by BraisGabin** [comment]
Wow, this proposal was flying under the radar! Those are really good suggestions. 
Could you send a PR / PRs with them?

**Comment by Eygonb** [comment]
Hi, if no one is planning to work on this, can I start?

**Comment by atulgpt** [comment]
<img width="999" alt="Screenshot 2023-09-12 at 3 44 58 PM" src="https://github.com/detekt/detekt/assets/11026203/0c5f3007-5295-4dbe-a6a3-5aca991f24c6">

Regarding #2 `I think forbiddenPatterns (default: '') is a bad name. patterns is plural. but it only takes a single pattern.` there are more similar plural names like in `allowedPatterns` in `ForbiddenComment`, and `allowedNames` in `UnusedParameter`, `UnusedPrivateMember` and `UnsuedPrivateProperty`

Also one more minor issue is that somewhere we are using `pattern`, in some places we are using `regex` and in some places we are using `names` to denote regex patterns

**Comment by BraisGabin** [comment]
> Hi, if no one is planning to work on this, can I start?

@Eygonb sorry for the really late responde. Yes, you can go for it :)

> Regarding numbere 2 `I think forbiddenPatterns (default: '') is a bad name. patterns is plural. but it only takes a single pattern.` there are more similar plural names like in `allowedPatterns` in `ForbiddenComment`, and `allowedNames` in `UnusedParameter`, `UnusedPrivateMember` and `UnsuedPrivateProperty`
> 
> Also one more minor issue is that somewhere we are using `pattern`, in some places we are using `regex` and in some places we are using `names` to denote regex patterns

If we end up with a good way to implement #6075 this issue should be fixed. But for now I think that remove the `s` from `patterns` is a good call.

**Comment by 3flex** [comment]
Which of these still need action? If we're renaming config parameters that should be done in 2.0 to avoid any breaking changes in later releases.

**Comment by marschwar** [comment]
Good point. I think:

*  the documentation here really could be improved upon. 
* Also I suggest we remove the `forbiddenPatterns` altogether. The glob should be sufficient and generally is easier to maintain.
* `imports` should be renamed to `forbiddenImports`

I can give it a shot, if there is agreement.

**Comment by BraisGabin** [comment]
Agree with your first and third point. But I think that we shouldn't remove `forbiddenPatterns`. It could be helpful for strange use cases that maybe need it. But I would rename it in any case.

#6075 could fix this second point but that issue doesn't move for really long time.

**Comment by marschwar** [comment]
I looked through public github repos that are using the `forbiddenPatterns` and found a few instances where the configuration could not be expressed with `forbiddenImports` alone. All of them were trying to exclude certain parts. For example I found:

```yaml
 ForbiddenImport:
    active: true
    # We forbid to import anything under androidx.compose.material because we should import from material3
    # We use a negative lookahead to allow for those classes we know are not on material3 and thus can be imported from material
    # e.g. (?!(Allowed|AlsoAllowed|AnotherAllowed)$)
    forbiddenPatterns: "androidx\\.compose\\.material\\.(?!(icons\\..*)$).+"
```

If we were to add a property `allowedImports`, this could be much more easily expressed as:

```yaml
 ForbiddenImport:
    active: true
    forbiddenImports:
      - value: 'androidx.compose.material.*'
        reason:  We forbid to import anything under androidx.compose.material because we should import from material3
    allowedImports:
      - 'androidx.compose.material.icons.*'
```

This is basically the suggestion from #6536.
